crates/floresta-wire/src/p2p_wire/peer.rs

```diff
@@ -450,16 +494,19 @@ impl<T: Transport> Peer<T> {
         Ok(())
     }
     #[allow(clippy::too_many_arguments)]
+    // change the name as transport is removed
     pub fn create_peer_from_transport(
-        stream: T,
+        stream: TcpStream,
```

Comment:
Hey could you explain more, which 2 functions are you talking about. I guess want to remove the stream from create_peer and handle it in create_actor? could you please elaborate.

Reply:
I mean `create_peer_from_transport`  and `create_outbound_connection` can become just one function that doesn't handle connection logic.

Reply:
@Davidson-Souza , I have created `create_peer` function instead and created tcpstream actor in `node.rs` itself. Let me know if this is what you meant or any other updates needed here.

Reply:
Exactly that! The actual connection logic now should be deferred to the actor. So we can have multiple actors without modifying the peer struct. I think this will help a lot with e.g., #148 and even integrating [arti](https://gitlab.torproject.org/tpo/core/arti/-/tree/main) in the future.